osu.Game/Overlays/IExpandableControl.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Overlays
+{
+    /// <summary>
+    /// An interface for UI controls with the ability to expand/contract.
+    /// </summary>
+    public interface IExpandableControl : IExpandable
+    {
+        /// <summary>
+        /// Returns whether the UI control is currently in a dragged state.
+        /// </summary>
+        bool IsControlDragged { get; }
```

Comment:
why is this an interface member even

Reply:
Have mentioned about it in https://github.com/ppy/osu/pull/16574#discussion_r790208831 already but I guess we can just discuss this here now that it's been dragged all the way up.

`ExpandingControlContainer` shouldn't contract when the user is dragging the UI control outside of the container's area (which would lose hover), but with implementations like `SettingsSlider` and `ExpandableSlider`, the true `IsDragged` state would reside in the underlying UI control wrapped in these components, and I can't really find a way to retrieve them without exposing it via interface.

Open for further opinions, I think we can make this naturally supported is by making `MosueEventManager` support dragging multiple drawables and make such wrapper components override `OnDragStart` and return true if their underlying control is in a dragged state.

Reply:
the point that if (and that's a *big* if) this should even exist, it should not be talking about "drag". that just shatters any semblance of abstraction brought by this interface. the actual *goal* of checking this - as i understand it - is to make sure that the control doesn't contract because something is being done on it. so an *actual* name for this (again, even if this is to exist) would be something like `StayExpanded` or similar.

Reply:
Well do you think these would no longer exist? I'm not sure about that even with the composition proposal.

I'm fine with renaming it to `StayExpanded`, I suppose that makes more sense.

Reply:
yes, as stated before i would rather have this not exist at all. if it is deemed *strictly necessary* to keep this interface in the discussion above then i'd be content with renaming.

Reply:
I haven't got to the point of understanding why this is required, but do agree that it should be renamed to `StayExpanded`, or `CanBeContracted` might work better? I guess with xmldoc either should be okay.

I'd also propose for simplicity that this is merged with the `IExpandable` interface unless there's reason not to.

Reply:
I've renamed it to `ShouldBeExpanded` and reworded the xmldoc to better explain the purpose of the property and also included the reason for its existence in the remarks section.